packages/elements/src/v2/src/components/image-cropper/handlers.ts

```diff
@@ -31,14 +33,20 @@ export const drawCanvasAfterCrop = ({
 
   let outputWidth
   let outputHeight
-
-  // get width or height, based on ratio
-  if (cropRatio > 1) {
-    outputWidth = originWidth
-    outputHeight = outputWidth / cropRatio
+  // if not set resizeDimensions, calculate from original width & height
+  if (!resizeDimensions) {
+    // get width or height, based on ratio
+    if (cropRatio > 1) {
+      outputWidth = originWidth
+      outputHeight = outputWidth / cropRatio
+    } else {
+      outputHeight = originHeight
+      outputWidth = outputHeight * cropRatio
+    }
   } else {
```

Comment:
```
if (! resizeDimensions && cropRatio > 1) {
       outputWidth = originWidth
      outputHeight = outputWidth / cropRatio
}
if(!resizeDimensions && cropRatio <= 1) {
       outputHeight = originHeight
      outputWidth = outputHeight * cropRatio
}
if(resizeDimensions) {
outputWidth = resizeDimensions.width
```
    outputHeight = resizeDimensions.height
}

Reply:
You could extract this into a function, and use `if` flow as above